### DISCUSSION TITLE
Unable to turn heater on/off

### DISCUSSION THREAD

**Discussion by FlaMike** [tagyoureit/nodejs-poolController#630] [Q&A]
I have an Easytouch 2 4 controller and I am running version 7.7.0 of the poolController.  I am also running Home Assistant and using the njsPC-HA integration to bring and control the pool from Home Assistant.  Everything has been working flawlessly for months.

A couple of days ago, for the first time, I tried to turn on the pool heater from Home Assistant.  I am using the climate.pool_heater entity in njsPC-HA.  I can change the heater temperature setpoint from HA without any problem and the change is reflected correctly on the Easytouch controller.

However, when I try to turn the heater on from HA, it immediately changes back to off.  In the poolControllerDashboard web interface, the heater is displayed as a virtual circuit but I cannot turn it on or off.  The heater does appear under heaters in the Dashboard and is present as a virtual circuit.

Can you kindly let me know what I need to do in order to use the heater via the dashboard and hopefully HA?  I can control it from the Easytouch controller, but I'd really like to be able to use the poolController instead of the Easytouch.

Thanks for your help!

![image](https://user-images.githubusercontent.com/41341956/201204387-861dc85d-e7f9-4aa9-8c47-db63f0f6aec1.png)

**Comment by rstrouse** [comment]
What heat mode do you have assigned in dashPanel?  To assign a heat mode tap on the heat mode for the body in the main dashboard.  This should include an option for Heater or Off.  When the water temp is lower than your set point it should turn the heater on.  Virtual circuits are set by the controller and can be used for triggers.  They are not for user control of the device.

**Reply by FlaMike** [reply]
Thank you for your response.  The heat mode was off.  When I toggle it to Heater in the dashboard and set the setpoint above the current water temperature, the heater works as expected.

I generally have the heater off unless I'm planning to swim and the water is too cold--I don't shutdown the pool in the winter.  The freeze protect does come on from occasionally, so it  is possible the water temperature may get below 40 degrees, the lowest heater setpoint.  I don't think I've seen the water temperature dip below 40, but I suppose it could happen.  (I live in north Florida, so hard freezes and extended cold periods are very rare here.)

Historically, the setpoint is at the desired water temperature & I just turn on the heater through Home Assistant.  Is there a way to set the heat mode though Home Assistant's UI or do I need to use the Dashboard?  I do not see an entity that's specific to heat mode in Home Assistant.

Thanks again for your help, swift response, and fabulous poolController package!

**Reply by tagyoureit** [reply]
I would have Crewski answer the question about how the HA integration handles this, but it should definitely be possible.

And fwiw, the freeze protect is based on the air temperature not water temperature.  When the air temp gets <32F (maybe it has a small buffer at 33/34, I can't recall), it will turn on any circuits that have freeze enabled.  The heater will never turn on automatically for freeze protection.  If you set the heatmode to heater and setpoint to 40 than you can be assured the water will always be above 40 degrees but, again, that's a different concept than the freeze protect.  Hope that helps!  

And thanks for the comments.  :)

**Reply by FlaMike** [reply]
Understood.

I suppose I could set the water temp to 40 & keep my fingers crossed that the heater doesn't go on due to darned cold water in January.  Or I could flip the the heater mode to off via the Daashboard in those cold winter months when I'm unlikely to swim & flip it back to Heat mode in the swimming season & just change the setpoint as apropos.

I shall contact crewski as you suggest.

Thank you again for your help and all!!!
